comp/core/workloadmeta/collectors/internal/process/process_collector_test.go

```diff
@@ -148,29 +180,51 @@ func TestCreatedProcessesCollection(t *testing.T) {
 		expectedProcesses  map[int32]*workloadmeta.Process
 	}{
 		{
-			description:     "single new process",
+			description:     "single new process with unknown language",
+			configOverrides: map[string]interface{}{},
+			processesToCollect: map[int32]*procutil.Process{
+				proc3.Pid: proc3,
+			},
+			expectedProcesses: map[int32]*workloadmeta.Process{
+				proc3.Pid: workloadmetaProcess(proc3,
+					&languagemodels.Language{
+						Name: languagemodels.Unknown,
+					}, nil),
+			},
+		},
+		{
+			description:     "single new process with python",
```

Comment:
I'm thinking we should try to test the language detection in a separate test than existing ones. The tests are starting to get quite large for "unit" tests.